### ISSUE TITLE
Unable to resolve type of interface field

### ISSUE THREAD

**Issue by cosmicbuffalo** [rmosolgo/graphql-ruby#1958]
So I'm trying to set up a schema with various interfaces that will resolve to different types based on context. I'm running into an issue where `field` definitions with interfaces aren't recognized in the schema and `resolve_type` is never getting called. Here's an example of what I'm talking about:

```ruby
module Interfaces
  module Customer
    include GraphQL::Schema::Interface
  
    field :id, ID, null: false
    field :email, String, null: false
  end
end
```
There's my interface ^

```ruby
class BaseCustomerType < GraphQL::Schema::Object
  implements Interfaces::Customer
end
```
There's my type I want to resolve to ^

```ruby
class QueryType < GraphQL::Schema::Object
  field :customer, Interfaces::Customer, null: false
  def customer
    Customer.last
  end
end
```
There's my query ^

```ruby
class Schema < GraphQL::Schema
  query QueryType
  context CustomContext

  def self.resolve_type(type, obj, ctx)
    # no matter what I do here, this is never called
    BaseCustomerType
  end
end
```

With all this setup, every time I try to execute a query I get an error like `"Field 'customer' doesn't exist on type 'QueryType'"`

Am I doing something wrong here, or is this a bug?

Looking at these docs (http://graphql-ruby.org/type_definitions/interfaces.html), it seems like this is supposed to work?

**Comment by chuckremes2** [comment]
According to the [schema docs](http://graphql-ruby.org/schema/definition) and the interface docs you point to, this should work.

I suggest making a failing test for this and using that to solve the problem.

**Comment by cosmicbuffalo** [comment]
I was able to somewhat figure out this issue btw

Turns out I needed to define orphan_types on the schema, and then resolve_type was called. There should probably be some clearer documentation on this. I found that it was mentioned here: http://graphql-ruby.org/type_definitions/interfaces.html#orphan-types, but these docs don't say anything about the relationship orphan_types and the resolve_type method. I'll make a pr to potentially update the docs a bit.
